src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -575,10 +590,20 @@ private static ISettings LoadSettingsGivenSettingsFiles(List<SettingsFile> setti
                     throw new InvalidOperationException(message);
                 }
 
-                userSpecificSettings = ReadSettings(rootDirectory, configFileName, settingsLoadingContext: settingsLoadingContext);
+                yield return ReadSettings(rootDirectory, configFileName, settingsLoadingContext: settingsLoadingContext);
             }
+        }
+
+        private static string GetUserSettingsDirectory(string rootDirectory, bool useTestingGlobalPath)
+        {
+            return useTestingGlobalPath
+                ? Path.Combine(rootDirectory, "TestingGlobalPath")
```

Comment:
Can we also make the string literal a constant?

```suggestion
                ? Path.Combine(rootDirectory, ConfigurationConstants.TestingGlobalPath);
```

Reply:
That's old code, mostly used for testing tbh. 

There's no contract that's important there.